homeassistant/components/bluetooth/manager.py

```diff
@@ -75,12 +77,18 @@ def _async_trigger_matching_discovery(
         self, service_info: BluetoothServiceInfoBleak
     ) -> None:
         """Trigger discovery for matching domains."""
+        discovery_key = discovery_flow.DiscoveryKey(
+            domain=DOMAIN,
+            key=service_info.address,
+            version=1,
+        )
         for domain in self._integration_matcher.match_domains(service_info):
```

Comment:
`self._integration_matcher.match_domains(service_info)` is most likely to return an empty set here.  We could guard creation of the discovery key here... need to check how often this actually runs though

Reply:
It is quite rare. so probably not worth optimizing for